test/built-ins/RegExp/S15.10.2.12_A2_T1.js

```diff
@@ -0,0 +1,534 @@
+// Copyright 2009 the Sputnik authors.  All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+info: |
+    The production CharacterClassEscape :: S evaluates by returning
+    the set of all characters not included in the set returned by
+    CharacterClassEscape :: s
+es5id: 15.10.2.12_A2_T1
+description: WhiteSpace
```

Comment:
I had to restore this whole file as it tests for `/\S+/g` which I'm not exactly doing in my new tests. The other files will check for single `\S` occurrences.

Reply:
I want to do that but I'd also like to test it on composed strings, which reflects what's being done in the legacy file. For this case, I'd rather work in another follow up as I have some other priorities to work I don't want to block this part. WDYT?

Reply:
There’s no rush to merging this PR, IMHO. I’d rather leave this open until it’s resolved, otherwise it’s easy to forget.

Reply:
Yeah, but I want to create something more consistent anyway for the general escapes, including \w, \W, \s, \S, etc.

I'll try to the requested parts done here, as a follow up I'll check how I can generate some tests from unicode data, etc.

Reply:
To generate tests based on Unicode data, perhaps node-unicode-data packages (e.g. https://github.com/mathiasbynens/unicode-11.0.0) come in useful.

Reply:
…and in case it helps, here’s how I generate the code point lists for `\d`, `\w`, etc. for regexpu-core: https://github.com/mathiasbynens/regexpu-core/blob/master/scripts/character-class-escape-sets.js

Output: https://github.com/mathiasbynens/regexpu-core/blob/master/data/character-class-escape-sets.js

Reply:
that's great, @mathiasbynens! I'll reuse this work for sure.